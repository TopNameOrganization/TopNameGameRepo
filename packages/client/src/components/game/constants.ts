export const TileSize = 32;
export enum Tile {
  Empty = 0,
  Brick = 1,
  Concrete = 2,
  Stair = 3,
  Rope = 4,
  Bonus = 5,
  Enemy = 6,
  Player = 7,
  Out = 100,
}

export const VELOCITY = 200;
export enum Orientation {
  Left = 0,
  Right = 1,
}
export enum RunnerAction {
  Stay = 0,
  Fall = 1000,
  MoveLeft = 37,
  MoveUp = 38,
  MoveRight = 39,
  MoveDown = 40,
}

export enum AnimationPhase {
  Stay = 0,
  Run = 1,
  Climb = 2,
  Hang = 3,
  Fall = 4,
}
