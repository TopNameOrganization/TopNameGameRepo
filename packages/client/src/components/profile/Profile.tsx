import * as React from 'react';
import Box from '@mui/material/Box';
import ProfileAvatar from "./ProfileAvatar";
import {ProfileForm} from "./ProfileForm";
import {PasswordForm} from "./PasswordFrom";
import Button from "@mui/material/Button";

export const Profile = () => {
    const [showForm, setShowForm] = React.useState<boolean>(false);
    const [showProfileForm, setShowProfileForm] = React.useState<boolean>(true);
    const [showPasswordForm, setShowPasswordForm] = React.useState<boolean>(false);

    const useForm = () => {
        setShowForm(false);
        setShowProfileForm(true);
        setShowPasswordForm(false);
    };

    const useProfileForm = () => {
        setShowForm(true);
        setShowProfileForm(true);
        setShowPasswordForm(false);
    };

    const usePasswordForm = () => {
        setShowForm(true);
        setShowProfileForm(false);
        setShowPasswordForm(true);
    };

    return (
        <Box
            sx={{
                marginTop: 8,
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
            }}
        >
            <ProfileAvatar />
            {
                showProfileForm &&
                <ProfileForm noUseForm={useForm} isDisabled={showForm}/>
            }
            {
                showForm &&
                showPasswordForm &&
                <PasswordForm noUseForm={useForm} />
            }
            {
                !showForm &&
                <Button
                    onClick={useProfileForm}
                    type="button"
                    fullWidth
                    variant="contained"
                    sx={{mt: 5, mb: 2}}
                >
                    Edit Profile
                </Button>
            }
            {
                !showForm &&
                <Button
                    onClick={usePasswordForm}
                    type="button"
                    fullWidth
                    variant="contained"
                    sx={{mt: 1, mb: 2}}
                >
                    Change Password
                </Button>
            }
        </Box>
    );
}
