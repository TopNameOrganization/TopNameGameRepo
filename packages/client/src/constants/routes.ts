export const ROUTES = {
  root: '/',
  home: '/home',
  game: '/game',
  login: '/login',
  signup: '/signup',
  profile: '/profile',
  leaderBoard: '/leader-board',
  forum: '/forum',
  forumCreate: '/forum/create',
  forumDetail: '/forum/:id',
}
