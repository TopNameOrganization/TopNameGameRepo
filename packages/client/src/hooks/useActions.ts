import {useDispatch} from "react-redux";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from '../store/actions/index'

export const useActions = () => {
  const dispatch = useDispatch()
  return bindActionCreators(Actions, dispatch)
}
