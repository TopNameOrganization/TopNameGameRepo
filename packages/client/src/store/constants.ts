import {UserState} from "./types";

export const initialState: UserState = {
  user: {
    id: null,
    first_name: '',
    second_name: '',
    display_name: '',
    login: '',
    email: '',
    phone: '',
    avatar: ''
  },
  loading: false,
  error: null
}
