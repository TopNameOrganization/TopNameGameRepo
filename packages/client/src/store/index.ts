import { rootReducer, RootState } from './reducers'
import { configureStore } from "@reduxjs/toolkit";

export const setupStore = (initialState?: RootState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState: initialState
  });
}
