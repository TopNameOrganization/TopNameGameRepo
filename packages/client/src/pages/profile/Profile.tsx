import * as React from 'react';
import {createTheme, ThemeProvider} from '@mui/material/styles';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import Profile from "../../components/profile/Profile";

const theme = createTheme();

export default function ProfilePage() {
    return (
        <ThemeProvider theme={theme}>
            <Container component="main" maxWidth="xs">
                <Typography component="h1" variant="h5">
                    Profile page
                </Typography>
                <Profile />
            </Container>
        </ThemeProvider>
    );
}
