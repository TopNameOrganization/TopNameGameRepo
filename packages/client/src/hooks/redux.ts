import {TypedUseSelectorHook, useDispatch, useSelector} from "react-redux";
import {AppDispatch, RootState} from "../store/reducers";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from "../store/actions";

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const useAppActions = () => bindActionCreators(Actions, useAppDispatch())
