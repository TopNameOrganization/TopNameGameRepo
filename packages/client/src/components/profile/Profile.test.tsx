import * as renderer from 'react-test-renderer';
import { Profile } from './Profile';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import * as appHooks from "../../hooks/redux";
import { Provider } from "react-redux";
import { userInitialState } from "../../store/constants";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const useAppSelectorMock = jest.spyOn(appHooks, 'useAppSelector');

describe('Profile', () => {
    it('should renders profile correctly', () => {
        useAppSelectorMock.mockReturnValue(userInitialState)
        const store = mockStore(userInitialState)
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Profile />
                </Provider>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
