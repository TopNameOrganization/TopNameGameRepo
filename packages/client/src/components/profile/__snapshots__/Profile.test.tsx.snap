// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should renders profile correctly 1`] = `
<div
  className="MuiBox-root css-binzgt"
>
  <div
    className="MuiBox-root css-1142v5w"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular css-ghc07c-MuiAvatar-root"
    >
      <img
        alt="Nick name"
        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="https://ya-praktikum.tech/api/v2/resourcesundefined"
      />
    </div>
  </div>
  <form
    className="MuiBox-root css-164r41r"
    id="profileForm"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="first_name"
        id="first_name-label"
      >
        First name
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="first_name"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={true}
          id="first_name"
          name="first_name"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="second_name"
        id="second_name-label"
      >
        Second name
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="second_name"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={true}
          id="second_name"
          name="second_name"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="display_name"
        id="display_name-label"
      >
        Nick name
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="display_name"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={true}
          id="display_name"
          name="display_name"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="login"
        id="login-label"
      >
        login
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="login"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={true}
          id="login"
          name="login"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="email"
        id="email-label"
      >
        Email Address
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={true}
          id="email"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="phone"
        id="phone-label"
      >
        Phone
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="phone"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={true}
          id="phone"
          name="phone"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="tel"
          value=""
        />
      </div>
    </div>
  </form>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1p8cus0-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Edit Profile
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1tav473-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Change Password
  </button>
</div>
`;
